plugins/modules/system/pamd.py

```diff
@@ -737,7 +737,7 @@ def __str__(self):
             lines.append(str(current_line))
             current_line = current_line.next
 
-        if lines[1].startswith("# Updated by Ansible"):
+        if len(lines) > 1 and lines[1].startswith("# Updated by Ansible"):
             lines.pop(1)
 
         lines.insert(1, "# Updated by Ansible - " + datetime.now().isoformat())
```

Comment:
Hmm, if `len(lines) == 0`, this will insert the marker in line 1, where the next invocation of this module won't find it (since it looks in line 2). (I'm using 1-based line numbers in my comment ;) )

Reply:
Good catch, thanks for spotting it. Better solution coming out of the oven soon.

Reply:
Err... I wrote a solution handling all the cases (==0, ==1, >1), then I expanded the integration test to check on the 0-line case. It failed miserably. Then I read the code/docs more thoroughly and... *drum roll* the module, as-is, cannot perform any change in a file that does not contain any rule. All the states it has will perform actions on or around the rules resulting from searching the file for an existing pattern. If there is no match, there is no action => modified file is never written =>  that `__str()__` is never called.

I think that is a flaw in this file - not idempotent, and it fails if the file does not yet exists - but not something to be dealt with in this PR.

Reply:
Anyway, I changed that implementation so we always get the special ansible comment *before* any rule.